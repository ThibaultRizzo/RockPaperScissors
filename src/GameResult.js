import React, { Component } from 'react';
import { Result } from './Rules.js';

function TurnNumber(props) {
  const numberArray = Array.from(Array(props.turnNumber).keys());
  const turns = numberArray.map(
    number => <th className="turn-number">{number}</th>
  );
  return turns;
}
function MovesList(props) {
  let moves = 
  moves.push(props.moves.map(
    (move,i) => <td key={i} className={move.win}>{move.value}</td>
  ));
  return moves; 
}

class GameResult extends Component {
  constructor(props) {
    super(props);
    this.state({
      movesP1: [],
      movesP2: [],
    });
  }
  getResolveLine() {
    let result = null,
      turn = this.props.turn;
    if (!this.props.turn) {
      result = false;
    } else if (Result.WIN === turn.result) {
      result = <p>You won !</p>;
    } else if (Result.LOSE === turn.result) {
      result = <p>You lost !</p>;
    } else {
      result = <p>It's a draw !</p>;
    }
    return result;
  }

  render() {
    let moveP1, moveP2;
    let turnNumber = resolveTurnNUm
    p1moves
    let result = this.getResolveLine();
    if (this.props.turn && result) {
      moveP1 = this.props.turn.moveP1;
      moveP2 = this.props.turn.moveP2;
    } else {
      return false;
    }
    return (
      <table id="GameResult">
        <tr>
          <TurnNumber turnNumber={this.props.turnNumber}/>
        </tr>
        <tr>
          <MovesList player="P1" moves={this.state.movesP1}/>
        </tr>
        <tr>
        <MovesList player="P2" moves={this.state.movesP2}/>
        </tr>
      </table>
    );
  }
}

export default GameResult;
