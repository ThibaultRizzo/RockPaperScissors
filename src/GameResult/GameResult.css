#GameResult {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
}

.result-grid {
  flex: 0 0 70%;
  height: 50%;
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: center;
  box-shadow: #7d7d7d 0px 0px 31px 6px;
}

.turn-row {
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
  align-items: stretch;
  height: 100%;
  flex-grow: 1;
  text-align: center;
}

.turn-number {
  font-weight: bolder;
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
}

.player-1,
.player-2 {
  font-weight: bolder;
}

.player-2.win {
  color: red;
}

.player-2.lose {
  color: green;
}
