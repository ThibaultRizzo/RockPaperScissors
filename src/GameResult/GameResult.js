import React, { Component } from 'react';
import './GameResult.css';

function TurnRow(props) {
  let result = props.turn.result.toLocaleLowerCase();
  return (
    <div className="turn-row">
      <div className={result + ' player-1'}>{props.turn.moveP1}</div>
      <div className="turn-number">{props.turnCount + 1}</div>
      <div className={result + ' player-2'}>{props.turn.moveP2}</div>
    </div>
  );
}

class GameResult extends Component {
  constructor(props) {
    super(props);
    this.state = {
      movesP1: [],
      movesP2: []
    };
  }

  render() {
    if (!this.props.turnArray) {
      return false;
    }
    let turnRows = this.props.turnArray.map((move, i) => (
      <TurnRow key={i} turnCount={i} turn={move} />
    ));
    return (
      <div id="GameResult">
        <div className="result-grid">{turnRows}</div>
      </div>
    );
  }
}

export default GameResult;
