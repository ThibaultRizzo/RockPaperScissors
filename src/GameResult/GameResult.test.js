import React from 'react';
import ReactDOM from 'react-dom';
import GameResult from './GameResult';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
      const props = {
        "isGameFinished": false,
        "turnArray": [{
          "moveP1": "PAPER",
          "moveP2": "SCISSORS",
          "result": "LOSE"
        }, {
          "moveP1": "ROCK",
          "moveP2": "SCISSORS",
          "result": "WIN"
        }],
        "turnNumber": 2
      };
      const tree = renderer
        .create( < GameResult isGameFinished = {
            props.isGameFinished
          }
          turnArray = {
            props.turnArray
          }
          turnNumber = {
            props.turnNumber
          }
          />)
          .toJSON(); expect(tree).toMatchInlineSnapshot(`
<a
  className="normal"
  href="https://prettier.io"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Prettier
</a>
`);
console.log(tree.toJSON());
        });