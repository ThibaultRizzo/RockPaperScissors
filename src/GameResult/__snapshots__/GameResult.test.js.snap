// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="GameResult"
>
  <div
    className="result-grid"
  >
    <div
      className="turn-row"
    >
      <div
        className="lose player-1"
      >
        ROCK
      </div>
      <div
        className="turn-number"
      >
        1
      </div>
      <div
        className="lose player-2"
      >
        PAPER
      </div>
    </div>
    <div
      className="turn-row"
    >
      <div
        className="lose player-1"
      >
        PAPER
      </div>
      <div
        className="turn-number"
      >
        2
      </div>
      <div
        className="lose player-2"
      >
        SCISSORS
      </div>
    </div>
    <div
      className="turn-row"
    >
      <div
        className="draw player-1"
      >
        SCISSORS
      </div>
      <div
        className="turn-number"
      >
        3
      </div>
      <div
        className="draw player-2"
      >
        SCISSORS
      </div>
    </div>
  </div>
</div>
`;
